allennlp/models/reading_comprehension/dialog_qa.py

```diff
@@ -0,0 +1,408 @@
+import logging
+from typing import Any, Dict, List
+import numpy as np
+from overrides import overrides
+import torch
+import torch.nn.functional as F
+from torch.nn.functional import nll_loss
+
+from allennlp.common import squad_eval
+from allennlp.data import Vocabulary
+from allennlp.models.model import Model
+from allennlp.modules import Seq2SeqEncoder, TimeDistributed, TextFieldEmbedder
+from allennlp.modules.matrix_attention.linear_matrix_attention import LinearMatrixAttention
+from allennlp.modules.input_variational_dropout import InputVariationalDropout
+from allennlp.nn import InitializerApplicator, util
+from allennlp.training.metrics import Average, BooleanAccuracy, CategoricalAccuracy
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@Model.register("dialog_qa")
+class DialogQA(Model):
+    """
+    This class implements modified version of BiDAF
+    (with self attention and residual layer, from Clark and Gardner ACL 17 paper) model as used in
+    Question Answering in Context (EMNLP 2018) paper for baseline.
+    In this set-up, a single instance is a dialog, list of question answer pairs.
+    Parameters
+    ----------
+    vocab : ``Vocabulary``
+    text_field_embedder : ``TextFieldEmbedder``
+        Used to embed the ``question`` and ``passage`` ``TextFields`` we get as input to the model.
+    phrase_layer : ``Seq2SeqEncoder``
+        The encoder (with its own internal stacking) that we will use in between embedding tokens
+        and doing the bidirectional attention.
+    span_start_encoder : ``Seq2SeqEncoder``
+        The encoder that we will use to incorporate span start predictions into the passage state
+        before predicting span end.
+    span_end_encoder : ``Seq2SeqEncoder``
+        The encoder that we will use to incorporate span end predictions into the passage state.
+    dropout : ``float``, optional (default=0.2)
+        If greater than 0, we will apply dropout with this probability after all encoders (pytorch
+        LSTMs do not apply dropout to their last layer).
+    num_context_answers : ``int``, optional (default=0)
+        If greater than 0, the model will consider previous question answering context.
+    mask_lstms : ``bool``, optional (default=True)
```

Comment:
Did you actually try this?  I saw some speed up with very early versions of allennlp, but @nelson-liu failed to reproduce the speedup with current code.  Just curious.

Reply:
no i haven't tried it, just copy-and-pasted them... ;P